mavros_extras/src/plugins/joy_control.cpp

```diff
@@ -0,0 +1,266 @@
+/**
+ * @brief JoyControl plugin
+ * @file joy_control.cpp
+ * @author Francois Chataigner <chataign@gmail.com>
+ *
+ * @addtogroup plugin
+ * @{
+ */
+/*
+ * This file is part of the mavros package and subject to the license terms
+ * in the top-level LICENSE file of the mavros repository.
+ * https://github.com/mavlink/mavros/tree/master/LICENSE.md
+ */
+
+#include <mavros/mavros_plugin.h>
+#include <pluginlib/class_list_macros.h>
+
+#include <sensor_msgs/Joy.h>
+
+#include <mavros_msgs/SetMode.h>
+#include <mavros_msgs/CommandBool.h>
+#include <mavros_msgs/State.h>
+#include <mavros_msgs/SetMode.h>
+
+#include <thread>
+#include <mutex>
+using namespace std;
+
+////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+
+template<typename Service>
+bool call_service( const string& service_name, const typename Service::Request& req )
```

Comment:
Why not to use `ros::ServiceClient`?
[tutorial link](http://wiki.ros.org/ROS/Tutorials/WritingServiceClient%28c%2B%2B%29#roscpp_tutorials.2BAC8-Tutorials.2BAC8-WritingServiceClient.Writing_the_Client_Node)